torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -396,13 +399,80 @@ void Reducer::mark_variable_ready_sparse(VariableIndex index) {
     replica.contents = grad;
     // See Note [DDP Communication Hook]
     if (comm_hook_ == nullptr) {
-      replica.contents.div_(process_group_->getSize());
+      replica.contents.div_(divFactor_);
     }
     // The grad is modified in place and needs to be written back.
     return true;
   });
 }
 
+std::vector<std::vector<at::Tensor>> Reducer::getBucketTensors() const {
+ std::lock_guard<std::mutex> lock(mutex_);
+ std::vector<std::vector<at::Tensor>> bucketTensors;
+ for (const auto& bucket : buckets_) {
+  std::vector<at::Tensor> tensors;
+  tensors.reserve(bucket.replicas.size());
```

Comment:
since we claim that we only support one-process-per-device use case, should we error out here if `bucket.replicas` has more than one element?